tests/platformSetup/tofu/modules/openstack/main.tf

```diff
@@ -0,0 +1,183 @@
+}
+
+resource "openstack_networking_port_v2" "nic" {
+  name = local.nic_name
+  network_id = openstack_networking_network_v2.net.id
+  fixed_ip {
+    subnet_id = openstack_networking_subnet_v2.subnet.id
+  }
+  security_group_ids = [openstack_networking_secgroup_v2.sg.id]
+}
+
+resource "openstack_networking_secgroup_v2" "sg" {
+  name        = local.fw_name
+  depends_on  = [openstack_networking_router_v2.router]
+}
+
+resource "openstack_networking_secgroup_rule_v2" "sg_icmp" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  remote_ip_prefix  = "0.0.0.0/0"
+  protocol          = "icmp"
+  security_group_id = openstack_networking_secgroup_v2.sg.id
+}
+
+resource "openstack_networking_secgroup_rule_v2" "sg_ssh" {
+  description       = "ssh"
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  remote_ip_prefix  = "0.0.0.0/0"
+  protocol          = "tcp"
+  port_range_min    = 22
+  port_range_max    = 22
+  security_group_id = openstack_networking_secgroup_v2.sg.id
+}
+
+resource "openstack_compute_keypair_v2" "key" {
+  name = local.ssh_key_name
+  public_key = file(var.ssh_public_key)
+}
+
+resource "openstack_images_image_v2" "image" {
+  name = local.image_name
+  local_file_path = local.image
+  container_format = "bare"
+  disk_format = "raw"
+  min_disk_gb = 3
+  min_ram_mb = 512
+  properties = {
+    architecture = local.arch
+    cpu_arch = local.arch
+    hypervisor_type = "qemu"
+    image_description = "Garden Linux Platform Test"
+    distro = "gardenlinux"
+    os_distro = "gardenlinux"
+    os_admin_user = var.ssh_user
+    # setting this to "required" will give us "nova.exception.SecureBootNotSupported: Secure Boot is not supported by host"
+    os_secure_boot = local.feature_trustedboot ? "optional" : null
+    # setting a NVRAM template could be an option to boot with different OVMF_VARS.fd with PK/DB/DBX/KEK
+    # this is undocumented and not working as of OpenStack 2024.1
+    # the OVMF_VARS.fd and template would have to exist on the nova compute node
+    # os_nvram_template = local.feature_trustedboot ? "/usr/share/qemu/firmware/99-edk2-x86_64-secure-enrolled-gl.json" : null
+    hw_firmware_type = "uefi"
+    hw_tpm_version = local.feature_tpm2 ? "2.0" : null
+    hw_tpm_model= local.feature_tpm2 ? "tpm-tis" : null
+  }
+}
+
+resource "openstack_compute_instance_v2" "instance" {
+  name              = local.instance_name
+  flavor_name       = var.instance_type
+  # image_name        = "Garden Linux 1592.6"
+  image_id          = openstack_images_image_v2.image.id
+  key_pair          = openstack_compute_keypair_v2.key.name
+  network {
+    port = openstack_networking_port_v2.nic.id
+  }
+
+  config_drive      = true
+  # user_data = filebase64("vm-startup-script.sh")
+
+  user_data = <<-EOT
+#cloud-config
+users:
+  - name: ${var.ssh_user}
+    lock_passwd: false
+    passwd: $6$saltsalt$8659GkBs0jVaaGWWSWwC3tFAFJmSs7706M2bi0rUR2DE0rF8ypszUoAghSTcdvZz3r0hVg4882tKMLwMUogdD.
```

Comment:
What would prevent me from using this to login to openstack instances during platform tests?

Reply:
Knowing the password I guess ;)
But good catch, I can remove this.